File: 01 Web Development 101/05 Ruby Challenges/02_calculator/calculator.rb
#write your code here
def add(a,b)
    a+b
  end
  
  def subtract(a,b)
    a-b
  end
  
  def sum(a)
    return 0 if a == []
    a.inject(:+)
  end
  
  def multiply(a)
    return 0 if a == []
    a.inject(:*)
  end
  
  def power(a,b)
    a**b
  end
  
  def factorial(a)
    return 0 if a == 0
    factorial = 1
    (1..a).each do |i|
      factorial *= i
    end
    factorial
  end

File: 02 JavaScript/08 Testing Practice/code/calculator.js
const calculator = {
	add: (a, b) => a + b,
	subtract: (a, b) => a - b,
	multiply: (a, b) => a * b,
	divide: (a, b) => {
		return b === 0 ? null : a / b;
	},
};

export default calculator;
